### PR TITLE
Dapp allow developer to specify from Name associated with the sender

### PR DESCRIPTION
**Pull request by luisccastillo**
The sender name is set to be optional and passed through the developer secret 3 (options) using the field "senderTag". The senderTag must not be empty(the variable is rather optional, but not empty), and be between 3 and 20 characters long.  The senderTag length may be modified upon request but it's not recommended to let users have long names since email providers may flag us as spam.

### CODE REVIEW

File: dapp/tests/unit/validateInputs.test.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ describe('validateInputs function', () => {
       mailJetSender: 'sender@example.com',
       emailSubject: 'Test email',
       emailContent: 'This is a it email',
+      senderTag: undefined,
```
Comment (PierreJeanjacquot) [inline dapp/tests/unit/validateInputs.test.js:17]: you can just remove it
```suggestion
```

File: dapp/tests/unit/emailService.test.js [hunk 2]
```diff
@@ -9,6 +9,7 @@ describe('sendEmail', () => {
     const emailSubject = 'Test email';
     const emailContent = 'This is a test email.';
     const mailJetSender = 'sender@example.com';
+    const senderTag = undefined;
```
Comment (PierreJeanjacquot) [inline dapp/tests/unit/emailService.test.js:12]: this variable assignment is useless

File: dapp/tests/e2e/app.test.js [hunk 3]
```diff
@@ -65,6 +65,33 @@ describe('sendEmail', () => {
...
+      Error('"contentType" must be one of [text/plain, text/html]')
+    );
+  });
+  it('should fail if IEXEC_REQUESTER_SECRET_3 (options) senderTag is empty', async () => {
+    process.env.IEXEC_REQUESTER_SECRET_3 = '{"senderTag":""}';
+    await expect(() => start()).rejects.toThrow(
+      Error('"senderTag" is not allowed to be empty')
+    );
+  });
+  it('should fail if IEXEC_REQUESTER_SECRET_3 (options) senderTag lenght is less than 3 characters', async () => {
```
Comment (PierreJeanjacquot) [inline dapp/tests/e2e/app.test.js:80]: typo

```suggestion
  it('should fail if IEXEC_REQUESTER_SECRET_3 (options) senderTag length is less than 3 characters', async () => {
```

I suggest using [Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) extension in VSCode

**Review by PierreJeanjacquot — CHANGES_REQUESTED** [review]
Good job!
I think we should use `senderName` for the option name rather than `senderTag` which is easier to understand see also some minor remarks.

**Comment by luisccastillo** [comment]
Changes updated :)

**Review by PierreJeanjacquot — APPROVED** [review]
LGTM :+1:

**Pull request merged**